querydsl-apt/src/main/java/com/querydsl/apt/DefaultConfiguration.java

```diff
@@ -255,7 +255,9 @@ public DefaultConfiguration(
         } else {
             variableNameFunction = DefaultVariableNameFunction.INSTANCE;
         }
-        module.bind(CodegenModule.VARIABLE_NAME_FUNCTION_CLASS, variableNameFunction);
+
+        String generatedAnnotationClass = GeneratedAnnotationResolver.resolve(options.get(QUERYDSL_GENERATED_ANNOTATION_CLASS));
+        module.bind(CodegenModule.GENERATED_ANNOTATION_CLASS, generatedAnnotationClass);
```

Comment:
I think we could actually use type `Class<? extends Annotation>` here and make it ourselves a little easier in the codgen changes.